lib/SIL/SILFunctionType.cpp

```diff
@@ -209,8 +211,7 @@ CanSILFunctionType SILFunctionType::getWithDifferentiability(
 }
 
 CanSILFunctionType SILFunctionType::getWithoutDifferentiability() {
-  if (!isDifferentiable())
-    return CanSILFunctionType(this);
+  assert(isDifferentiable() && "Must be a differentiable function");
```

Comment:
I don't believe this assertion is the right thing. I remember writing a lot of `getWithoutDifferentiability()` without checking whether a type is differentiable or not. This is entirely different from `getDifferentiationParameterIndices() ` because the name `getDifferentiationParameterIndices()` implies the type is already differentiable, while `getWithoutDifferentiability()` does not.

Reply:
That makes sense, thanks! `getWithoutDifferentiability` works for both differentiable and non-differentiable function types.

I reverted this change to `getWithoutDifferentiability`. The patch now adds only one conservative assertion.